public/app/features/datasources/settings/DataSourceSettingsPage.tsx

```diff
@@ -38,55 +41,20 @@ export interface Props {
   updateDataSource: typeof updateDataSource;
   setIsDefault: typeof setIsDefault;
   dataSourceLoaded: typeof dataSourceLoaded;
+  initDataSourceSettings: typeof initDataSourceSettings;
+  testDataSource: typeof testDataSource;
   plugin?: GenericDataSourcePlugin;
   query: UrlQueryMap;
   page?: string;
-}
-
-interface State {
-  plugin?: GenericDataSourcePlugin;
-  isTesting?: boolean;
   testingMessage?: string;
   testingStatus?: string;
-  loadError?: any;
+  loadError?: Error | string;
```

Comment:
Nit: maybe we can do something about this? Not a big fan of having different types for the same state/prop.